src/vtcomposite.cpp

```diff
@@ -602,75 +615,86 @@ struct LocalizeWorker : Napi::AsyncWorker
         }
 
+        if (!worldview_key.empty())
+        {
+            fbuilder.add_property(worldview_key, worldview_val);
+        }
+
         fbuilder.commit();
     }
 
+    // returns a vector of requested worldviews that the feature exists in
+    static std::vector<std::string> worldviews_for_feature(std::vector<std::string> available_worldviews, std::vector<std::string> target_worldviews)
+    {
+        target_worldviews.emplace_back("all");
+
+        std::vector<std::string> matching_worldviews;
+        utils::intersection(available_worldviews, target_worldviews, matching_worldviews);
+        return matching_worldviews;
+    }
+
     void Execute() override
     {
         try
         {
-            vtzero::tile_builder tbuilder;
-            std::string language_key;
-            std::string language_key_prefixed;
-            if (!baton_data_->language.empty())
+            bool keep_every_worldview = true;
+            std::string incompatible_worldview_key;
+            std::string compatible_worldview_key;
+            std::vector<std::string> class_key_precedence;
+            bool keep_every_language = true;
+            std::vector<std::string> language_key_precedence;
+            if (baton_data_->return_localized_tile)
+            {
+                keep_every_worldview = false;
+                incompatible_worldview_key = baton_data_->worldview_property;
+                compatible_worldview_key = baton_data_->worldview_prefix + baton_data_->worldview_property;
+
+                class_key_precedence.push_back(baton_data_->class_prefix + baton_data_->class_property);
+                class_key_precedence.push_back(baton_data_->class_property);
+
+                keep_every_language = false;
+                for (auto const& lang : baton_data_->languages)
+                {
+                    language_key_precedence.push_back(baton_data_->language_property + "_" + lang);
+                    language_key_precedence.push_back(baton_data_->language_prefix + baton_data_->language_property + "_" + lang);
+                }
+                language_key_precedence.push_back(baton_data_->language_property);
+            }
+            else
             {
-                // {language_property}_{language} -> retain
-                language_key = baton_data_->language_property + "_" + baton_data_->language;
-                // {language_prefix}{language_property}_{language} -> drop
-                language_key_prefixed = baton_data_->language_prefix + language_key;
+                keep_every_worldview = true;
+                incompatible_worldview_key = baton_data_->worldview_prefix + baton_data_->worldview_property;
+                compatible_worldview_key = baton_data_->worldview_property;
+
+                class_key_precedence.push_back(baton_data_->class_property);
+
+                keep_every_language = true;
+                language_key_precedence.push_back(baton_data_->language_property);
             }
```

Comment:
Possibly you could make this more readable by encapsulating this if/else block in a function that returns the precedence arrays and name it something like `determine_precedences` or something. This whole function, by its nature, is dense and breaking it down could help make it more readable.

Reply:
👍 I'll leave function as is and break it down in a separate PR.

Reply:
🤔 since there might be things we need to change in this function to support scripts (details TBD), let's circle back to this after the scripts support SDD is done.

Reply:
👍